File: palindrome.py
# t - o(n), s - 0(n) 
class Solution:
    def longestPalindrome(self, s: str) -> int:
        hashset = set()
        answer = 0
        for i in range(0,len(s)):
            if s[i] in hashset:
                hashset.remove(s[i])
                answer += 2

            else:
                hashset.add(s[i])

        return answer + 1 if hashset else answer
        

File: Palindrome.cpp
class Solution {
    // Time Complexity : O(n)
    // Space Complexity: O(c) // constant as there at max 26 evtries to set 
public:
    int longestPalindrome(string s) {
        unordered_set<char> matchSet;
        int palindromeCount =0;
        for(int i=0;i<s.length();i++){
            if(matchSet.find(s[i]) != matchSet.end()){
                palindromeCount = palindromeCount+2;
                matchSet.erase(s[i]);
            }
            else  matchSet.insert(s[i]);
        }
        if (!matchSet.empty()){
            palindromeCount++;
        }
        return palindromeCount;


    }
};